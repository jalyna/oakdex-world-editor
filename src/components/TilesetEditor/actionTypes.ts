export const UPLOAD_TILESET = 'UPLOAD_TILESET'
export const UPDATE_TILESET = 'UPDATE_TILESET'
export const RESET_TILESET = 'RESET_TILESET'
export const CHANGE_TAB = 'CHANGE_TAB'
export const CHANGE_CURRENT_COORDINATES = 'CHANGE_CURRENT_COORDINATES'
export const REMOVE_CURRENT_COORDINATES = 'REMOVE_CURRENT_COORDINATES'
export const CHANGE_MOUSE_HOLD = 'CHANGE_MOUSE_HOLD'
