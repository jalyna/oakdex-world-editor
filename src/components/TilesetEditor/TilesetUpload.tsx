import * as React from 'react'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import styled from 'styled-components'

import { DEFAULT_FONT } from 'shared/theme'

import { UPLOAD_TILESET } from './actionTypes'

const StyledWrapper = styled.div`
  font-family: ${DEFAULT_FONT};
`

interface TilesetUploadProps {
  onChangeFile: () => void
}

function mapDispatchToProps (dispatch: Dispatch) {
  return {
    onChangeFile: (e: React.FormEvent<HTMLInputElement>) => {
      if (e.currentTarget.files && e.currentTarget.files[0]) {
        const file = e.currentTarget.files[0]
        if (file.name.indexOf('.json') >= 0) {
          alert('JSON is not possible yet')
        } else {
          const fr = new FileReader()
          const img = new Image()

          fr.addEventListener('load', (e) => {
            img.src = (fr.result as string)
            img.addEventListener('load', () => {
              console.log('woopwoop', img.naturalWidth, img.naturalHeight)
              dispatch({
                type: UPLOAD_TILESET,
                data: {
                  imageBase64: fr.result as string,
                  width: img.naturalWidth / 16,
                  height: img.naturalHeight / 16
                }
              })
            })
          })

          fr.readAsDataURL(e.currentTarget.files[0])
        }
      }
    }
  }
}

function TilesetUpload ({ onChangeFile }: TilesetUploadProps) {
  return (
    <StyledWrapper>
      <input type='file' onChange={onChangeFile} />
    </StyledWrapper>
  )
}

export default connect(
  () => { return {} },
  mapDispatchToProps
)(TilesetUpload)
