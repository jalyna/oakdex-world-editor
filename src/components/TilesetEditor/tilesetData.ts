import { Reducer } from 'redux'

import { UPLOAD_TILESET } from './actionTypes'

export interface Tileset {
  width: number,
  height: number,
  imageBase64: string
}

const tilesetData: Reducer<Tileset> = (state: Tileset | null = null, action): Tileset | null => {
  switch (action.type) {
    case UPLOAD_TILESET:
      return {
        ...action.data
      }
    default:
      return state
  }
}

export default tilesetData
