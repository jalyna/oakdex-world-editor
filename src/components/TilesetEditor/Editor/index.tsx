import * as React from 'react'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'

import { Tileset } from '../reducers/tilesetData'

interface EditorProps {
  tilesetData: Tileset
}

function mapStateToProps ({ tilesetData }: any) {
  return {
    tilesetData
  }
}

function Editor ({ tilesetData }: EditorProps) {
  return <div>EDITOR {tilesetData.title}</div>
}

export default connect(
  mapStateToProps,
  (d: Dispatch) => { return {} },
)(Editor)
