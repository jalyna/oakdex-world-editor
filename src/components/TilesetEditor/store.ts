import { combineReducers, createStore, StoreEnhancer } from 'redux'

import tilesetData from './tilesetData'

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION__: () => StoreEnhancer | null
  }
}

window.__REDUX_DEVTOOLS_EXTENSION__ = window.__REDUX_DEVTOOLS_EXTENSION__ || null

const debug = window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : null

const reducer = combineReducers({
  tilesetData
})

export default createStore(reducer, {}, debug)
