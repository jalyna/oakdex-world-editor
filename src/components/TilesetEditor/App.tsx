import * as React from 'react'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'

import { Tileset } from './reducers/tilesetData'
import Editor from './Editor'

interface AppProps {
  tilesetData: Tileset | null
}

function mapStateToProps ({ tilesetData }: any) {
  return {
    tilesetData
  }
}

function App ({ tilesetData }: AppProps) {
  if (tilesetData) {
    return <Editor />
  } else {
    return null
  }
}

export default connect(
  mapStateToProps,
  (d: Dispatch) => { return {} },
)(App)
