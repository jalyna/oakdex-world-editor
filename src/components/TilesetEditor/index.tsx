import * as React from 'react'
import { Provider } from 'react-redux'

import { GREY_50 } from 'shared/theme'

import store from './store'

export default function TilesetEditor ({}) {
  return (
    <Provider store={store}>
      Hello Redux
    </Provider>
  )
}
