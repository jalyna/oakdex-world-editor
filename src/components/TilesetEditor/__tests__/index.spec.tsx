import * as React from 'react'
import { shallow } from 'enzyme'
import TilesetEditor from '..'

describe('<TilesetEditor>', () => {
  it('renders', () => {
    const editor = shallow(<TilesetEditor />)
    expect(editor).toMatchSnapshot()
  })
})
