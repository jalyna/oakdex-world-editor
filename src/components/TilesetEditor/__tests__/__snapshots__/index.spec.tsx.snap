// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TilesetEditor> renders 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(AppComponent) />
</Provider>
`;
